WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsViewModel.swift

```diff
@@ -40,7 +42,9 @@ final class ProductSettingsViewModel {
     }
 
     func hasUnsavedChanges() -> Bool {
-        if product.statusKey != productSettings.status.rawValue {
+        if product.statusKey != productSettings.status.rawValue ||
+            product.featured != productSettings.featured ||
+            product.catalogVisibilityKey != productSettings.catalogVisibility.rawValue {
```

Comment:
non-blocking nit: maybe we could create a `ProductSettings` convenience initializer that takes a `Product`, and implement `Equatable` for `ProductSettings` if it hasn't, then just compare `ProductSettings(from: product) != productSettings` here?

Reply:
This is a good suggestion! I implemented it 💯